packages/shared/src/h5t.ts

```diff
@@ -0,0 +1,53 @@
+/* ------------------- */
+/* ---- H5T ENUMS ---- */
+
+// https://docs.hdfgroup.org/hdf5/v1_14/_h5_tpublic_8h.html#title3
+
+export enum H5T_CLASS {
+  INTEGER = 0,
+  FLOAT = 1,
+  TIME = 2,
+  STRING = 3,
+  BITFIELD = 4,
+  OPAQUE = 5,
+  COMPOUND = 6,
+  REFERENCE = 7,
+  ENUM = 8,
+  VLEN = 9,
+  ARRAY = 10,
+}
+
+export enum H5T_ORDER {
+  LE = 0,
+  BE = 1,
+  VAX = 2,
+  MIXED = 3,
+  NONE = 4,
+}
+
+export enum H5T_SIGN {
+  NONE = 0, // unsigned
+  SIGN_2 = 1, // signed
+  NSGN = 2,
+}
+
+export enum H5T_CSET {
+  ASCII = 0,
+  UTF8 = 1,
+}
+
+/* ---------------------- */
+/* ---- H5T MAPPINGS ---- */
+
+export const H5T_TO_ENDIANNESS = {
+  [H5T_ORDER.LE]: 'little-endian',
+  [H5T_ORDER.BE]: 'big-endian',
+  [H5T_ORDER.VAX]: 'VAX',
+  [H5T_ORDER.MIXED]: 'mixed',
+  [H5T_ORDER.NONE]: 'none',
+} as const;
+
+export const H5T_TO_CHAR_SET = {
+  [H5T_CSET.ASCII]: 'ASCII',
+  [H5T_CSET.UTF8]: 'UTF-8',
+} as const;
```

Comment:
These are the mappings from H5T enums to readable strings. Could have used arrays, but this felt more explicit. The `as const` are important to generate the union types `CharSet` and `Endianness` in `hdf5-models.ts`.

As mentioned, I did not add error/reserved members to the enums, since the idea is that if such a value is received from a provider, the mapping will return `undefined` - e.g. `H5T_TO_ENDIANNESS[-1 as H5T_ORDER] => undefined`.

To make this safe, I added `| undefined` in the relevant places in `hdf5-models.ts`.